### ISSUE TITLE
A-Painter 404s when URL does not have a trailing slash

### ISSUE THREAD

**Issue by chrisplatorres** [exokitxr/exokit#1025]
**Describe the bug**
What happened, and what did you expect to happen?
 
running a-painter with a trailing slash as a tab - `node --experimental-worker . -h -t https://aframe.io/a-painter/` gives a normal log.

running a-painter without a trailing slash in URL as a tab, giving a lot of 404s - `node --experimental-worker . -h -t https://aframe.io/a-painter`
 

**Additional context**

I think a-painter treats it different when there is a trailing slash or not.

**System information:**

Windows 10
Built from source
`user-parallel`
Node v11.6.0
Npm v 6.5.0-next.0

**Comment by modulesio** [comment]
Without the `/`:

```
ErrorEvent {
  type: 'error',
  target: <script src="vendor/aframe-master.min.js"></script>,
  bubbles: false,
  cancelable: false,
  defaultPrevented: false,
  propagationStopped: false,
  currentTarget: <script src="vendor/aframe-master.min.js"></script>,
  message:
   'script src got invalid status code: 404 : vendor/aframe-master.min.js',
  stack:
   'Error: script src got invalid status code: 404 : vendor/aframe-master.min.js\n    at ownerDocument.defaultView.fetch.then.res (C:\\Users\\avaer\\Documents\\GitHub\\exokit\\src\\DOM.js:1700:35)\n    at process.internalTickCallback (internal/process/next_tick.js:77:7)' }
Error: script src got invalid status code: 404 : vendor/aframe-master.min.js
    at ownerDocument.defaultView.fetch.then.res (C:\Users\avaer\Documents\GitHub\exokit\src\DOM.js:1700:35)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
ErrorEvent {
  type: 'error',
  target:
   <script src="vendor/aframe-input-mapping-component.js"></script>,
  bubbles: false,
  cancelable: false,
  defaultPrevented: false,
  propagationStopped: false,
  currentTarget:
   <script src="vendor/aframe-input-mapping-component.js"></script>,
  message:
   'script src got invalid status code: 404 : vendor/aframe-input-mapping-component.js',
  stack:
   'Error: script src got invalid status code: 404 : vendor/aframe-input-mapping-component.js\n    at ownerDocument.defaultView.fetch.then.res (C:\\Users\\avaer\\Documents\\GitHub\\exokit\\src\\DOM.js:1700:35)\n    at process.internalTickCallback (internal/process/next_tick.js:77:7)' }
Error: script src got invalid status code: 404 : vendor/aframe-input-mapping-component.js
    at ownerDocument.defaultView.fetch.then.res (C:\Users\avaer\Documents\GitHub\exokit\src\DOM.js:1700:35)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
```

We had a hack to automatically append the `/` into the url passed from the shell, but that's error-prone.

**Comment by chrisplatorres** [comment]
I think there is now a fix in the `user-parallel` branch: 

https://github.com/exokitxr/exokit/pull/760/commits/ea1ccf8aa36fa4007ba1db8c16ba6b80f95f535d#diff-1fdf421c05c1140f6d71444ea2b27638R1183

**Comment by modulesio** [comment]
How does that fix work?

**Comment by chrisplatorres** [comment]
```js
url = url.replace(/\/?$/, '/');
```

- If the trailing slash exists, it is replaced with `/`.
- If the trailing slash does not exist, a `/` is appended to the end (to be exact: The trailing anchor is replaced with `/`).

An alternative to the regex could be:
```js
url += url.endsWith("/") ? "" : "/"
```

**Comment by modulesio** [comment]
Won't that break with any url like `index.html`?

**Comment by chrisplatorres** [comment]
Just tested, dding a trailing slash doesn't seem to affect local paths.

**Comment by modulesio** [comment]
Interesting, I remember commenting that fix out because local `file://` urls were busted.

**Comment by chrisplatorres** [comment]
Is there a repro case for that? I am running `studio` with that line un-commented and am able to open local files in Exokit.

**Comment by modulesio** [comment]
I think what I was doing is `node . --experimental-worker file://C:\some-file.html`

**Comment by chrisplatorres** [comment]
I am getting no issues running a site from above: `node --experimental-worker . file://C\\exokit\\hello.html`

**Comment by modulesio** [comment]
`file://C\\exokit\\hello.html` is that a valid path or a typo?

**Comment by chrisplatorres** [comment]
Yes I think so using git bash/MINGW64.

Tried again with `exobot.html` in `C:` and was able to load:
`$ node --experimental-worker . -t file:///exobot.html`

**Comment by chrisplatorres** [comment]
Closed by #1061.
